common/devdoc/interlocked_hl_requirements.md

```diff
@@ -165,7 +211,22 @@ MOCKABLE_FUNCTION_WITH_RETURNS(, INTERLOCKED_HL_RESULT, InterlockedHL_SetAndWake
 
 **SRS_INTERLOCKED_HL_02_019: [** `InterlockedHL_SetAndWake` shall succeed and return `INTERLOCKED_HL_OK`. **]**
 
-### InterlockedHL_SetAndWake
+### InterlockedHL_SetAndWake64
+```c
+MOCKABLE_FUNCTION_WITH_RETURNS(, INTERLOCKED_HL_RESULT, InterlockedHL_SetAndWake64, int64_t volatile_atomic*, address, int64_t, value)(INTERLOCKED_HL_OK, INTERLOCKED_HL_ERROR);
+```
+
+`InterlockedHL_SetAndWake64` set the value at `address` to `value` and signals the change of value in `address`. This can be commonly used with `InterlockedHL_WaitForValue64` to signal a waiting thread.
+
+**SRS_INTERLOCKED_HL_05_013: [** If address is NULL then InterlockedHL_SetAndWake64 shall fail and return INTERLOCKED_HL_ERROR. **]**
+
+**SRS_INTERLOCKED_HL_05_014: [** InterlockedHL_SetAndWake64 shall set value at address. **]**
+
+**SRS_INTERLOCKED_HL_05_015: [** InterlockedHL_SetAndWake64 shall wake a single thread listening on address. **]**
```

Comment:
>InterlockedHL_SetAndWake64 shall wake a single thread listening on address. [](http://example.com/codeflow?start=33&length=75)

Many requirements are same as their 32 bit counterpart, however some are worded slightly differently. I assume that functionally these requirements are going to produce the same code except for using 64 bit APIs instead of 32-bit ones. In that case, can we please use the same wording as the 32-bit requirement so that we have consistency?

Reply:
Only InterlockedHL_WaitForValue64 and InterlockedHL_WaitForNotValue64 is updated since some of the specs essentially meant the same thing and was repetitive.
Like SRS_INTERLOCKED_HL_01_005 doesn't convey anything and is just saying milliseconds would be used as timeout when in fact we can just name the input variable timeout_ms.

Reply:
Codes_SRS_INTERLOCKED_HL_01_008 and Codes_SRS_INTERLOCKED_HL_01_004 mean the same thing.
Codes_SRS_INTERLOCKED_HL_42_006 and Codes_SRS_INTERLOCKED_HL_42_003 mean the same thing.